File: solutions/cpp/bob/1/bob.cpp
#include "bob.h"

namespace bob {
    string hey(string hey_bob) {
        string bob_response{};

        if (hey_bob.empty() || regex_match(hey_bob, regex("^\\s*$"))) {
            bob_response = "Fine. Be that way!";
        } else if (regex_match(hey_bob, regex("[A-Z? %^*@#$(*^]+")) and hey_bob.back() == '?') {
            if (regex_search(hey_bob, regex("[a-zA-Z]")) == false) {
                bob_response = "Sure.";
            } else {
                bob_response = "Calm down, I know what I'm doing!";
            }
        } else if (regex_match(hey_bob, regex("[A-Z!0-9 ,%^*@#$(*^]+")) and hey_bob.back() != '?') {
            if (regex_search(hey_bob, regex("[a-zA-Z]")) == false) {
                bob_response = "Whatever.";
            } else {
                bob_response = "Whoa, chill out!";
            }
        } else if (hey_bob.back() == '?') {
            bob_response = "Sure.";
        } else if (hey_bob.find("bob") != string::npos) {
            bob_response = "Fine. Be that way!";
        } else if (hey_bob.back() == ' ') {
            if (hey_bob.find("?") != string::npos) {
                bob_response = "Sure.";
            } else {
                bob_response = "Whatever.";
            }
        } else {
            bob_response = "Whatever.";
        }
        
        return bob_response;
    }
}  // namespace bob

File: solutions/python/bob/1/bob.py
import re

def response(hey_bob):
    
    if hey_bob == "" or hey_bob.isspace() is True:
        bob_response = "Fine. Be that way!"
    elif hey_bob == hey_bob.upper() and hey_bob[-1] == "?":
        if re.search('[a-zA-Z]', hey_bob) == None:
            bob_response = "Sure."
        else:
            bob_response = "Calm down, I know what I'm doing!"
    elif hey_bob == hey_bob.upper() and hey_bob[-1] != "?":
        if re.search('[a-zA-Z]', hey_bob) == None:
            bob_response = "Whatever."
        else:
            bob_response = "Whoa, chill out!"
    elif hey_bob[-1] == "?":
        bob_response = "Sure."
    elif "bob" in hey_bob:
        bob_response = "Fine. Be that way!"
    elif hey_bob.endswith(' ') == True:
        if re.search('[?]', hey_bob) != None:
            bob_response = "Sure."
        else:
            bob_response = "Whatever."
    else:
        bob_response = "Whatever."

    return bob_response
